build_tools/cmake/iree_cc_library.cmake

```diff
@@ -197,6 +197,15 @@ function(iree_cc_library)
         ${_RULE_DEFINES}
     )
 
+    # If in BUILD_SHARED_LIBS mode, then we need to make sure that visibility
+    # is not hidden. We default to hidden visibility in the main copts so
+    # need to undo it here.
```

Comment:
Should we pull that out of the main copts and instead only set it if libraries aren't shared?

Reply:
Was debating that... LLVM has been in a tug-of-war over this in a similar way. I'd actually like to take it out entirely in favor of visibility preset... but more work: https://cmake.org/cmake/help/latest/prop_tgt/LANG_VISIBILITY_PRESET.html

Reply:
Added a TODO for the moment. There is some trickiness involved and I would like this PR to be NFC with respect to the current static builds.